sympy/solvers/diophantine.py

```diff
@@ -1040,13 +1040,13 @@ def _diop_quadratic(var, coeff, t):
 
         if D < 0:
             for solution in solns_pell:
-                s1 = P*Matrix([solution[0], solution[1]]) + Q
-                s2 = P*Matrix([-solution[0], solution[1]]) + Q
-                try:
-                    sol.add(tuple([as_int(_) for _ in s1]))
-                    sol.add(tuple([as_int(_) for _ in s2]))
-                except ValueError:
-                    pass
+                for X_i in [-solution[0], solution[0]]:
```

Comment:
It seems to me that this piece of code will create the solutions with different combinations of signs essentially in the same way as `permute=True` would work. The solutions with x and y swapped are added elsewhere, not in this loop.

I wanted to know if all solutions could be obtained without this part of code when the rest of the code is included and `permute=True` is set. I know that it will not currently give all solutions, but code that adds permuted signs should not be repeated.